### ISSUE TITLE
progressive mode has no effect

### ISSUE THREAD

**Issue by braindevices** [ansible-community/ansible-lint#1527]
##### Summary

progressive mode does not suppress old errors at all.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible 2.10.6

ansible-lint --version
ansible-lint 5.0.5 using ansible 2.10.6
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

ubuntu 20.04

##### STEPS TO REPRODUCE

1. introduce some error like following in a role:
```
- debug:
    var: ansible_host
```
2. `ansible-lint myplaybook.yml` which include that role, it raise an error.
3. commit the error to `add debug (test ansible-lint)`
4. change the role again by add some correct tasks, then run the `ansible-lint --progress myplaybook.yml`, it put an old-rev branch to master but still report the error.
5. commit the changes, then run the `ansible-lint --progress myplaybook.yml`, the old-rev stays same, but master is 1 commit ahead. But it still report the error.
6. repeat 4-5, always report error introduced by `add debug (test ansible-lint)`

```console (paste below)

```

##### Desired Behaviour

no error should detect

##### Actual Behaviour

`git log` :
```
commit fc9740008f198aaacd7cae63c8734f4ca1fbcbad (HEAD -> master)
Author: john@example.com
Date:   Fri Apr 9 14:27:40 2021 +0200

    empty commit Fri 09 Apr 2021 02:27:40 PM CEST

commit de48eaa129ddc89b556010f3e6ac8e8aaedd1a59
Author: john@example.com
Date:   Fri Apr 9 14:23:44 2021 +0200

    test2 add task without err

commit 1df9e090e3c75ac4609f3c5a1dd5f6de133855a5 (old-rev)
Author: john@example.com
Date:   Fri Apr 9 14:14:04 2021 +0200

    add debug (test ansible-lint)
```

```
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: myplaybook.yml
INFO     Executing syntax check on myplaybook.yml (0.39s)
INFO     Matches found, running again on previous revision in order to detect regressions
Preparing worktree (checking out 'old-rev')
Previous HEAD position was 1df9e09 add debug (test ansible-lint -p)
HEAD is now at de48eaa test2 add task without err
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: myplaybook.yml
INFO     Executing syntax check on myplaybook.yml (0.33s)
WARNING  Listing 1 violation(s) that are fatal
unnamed-task: All tasks should be named
roles/xxx/tasks/xxx.yml:33 Task/Handler: debug var=ansible_host

You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - unnamed-task  # All tasks should be named
Finished with 1 failure(s), 0 warning(s) on 6 file
```
Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:
- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
Progressive mode does not aim to hide known problems, it aims to avoid failing due to them. Here is a repository that is actively using progressive https://github.com/ansible-community/ansible-lint/blob/master/playbooks/eco.yml#L16-L19

If it would not work, zuul folks would be very upset.
